PhysicsTools/NanoAOD/python/jets_cff.py

```diff
@@ -36,9 +36,22 @@
      ),
 )
 
+from  PhysicsTools.PatAlgos.producersLayer1.jetUpdater_cfi import *
+from  PhysicsTools.PatAlgos.recoLayer0.jetCorrFactors_cfi import *
+updatedJets = updatedPatJets.clone(
+	addBTagInfo=False,
+	jetSource='slimmedJetsWithUserData',
+	jetCorrFactorsSource=cms.VInputTag(cms.InputTag("jetCorrFactors") ),
+)
+jetCorrFactors = patJetCorrFactors.clone(src='slimmedJetsWithUserData',
+    levels = cms.vstring('L1FastJet',
+        'L2Relative',
+        'L3Absolute'),
```

Comment:
do we need also the residuals?

Reply:
(can you also invert the order of the two, so that the cff introduces the modules in  the order they are run? it's easier for the reader)